src/natcap/invest/urban_flood_risk_mitigation.py

```diff
@@ -579,39 +569,39 @@ def _write_summary_vector(
         field_def.SetPrecision(11)
         target_watershed_layer.CreateField(field_def)
 
-    target_layer_defn = target_watershed_layer.GetLayerDefn()
-    for base_feature in source_aoi_layer:
-        feature_id = base_feature.GetFID()
-        target_feature = ogr.Feature(target_layer_defn)
-        base_geom_ref = base_feature.GetGeometryRef()
-        target_feature.SetGeometry(base_geom_ref.Clone())
-        base_geom_ref = None
+    target_watershed_layer.ResetReading()
+    for target_feature in target_watershed_layer:
+        # Target vector is SHP, where FIDs start at 0, but stats were
+        # generated based on GPKG reprojection, where FIDs start at 1.
+        # Therefore, we need to reference stats at SHP FID + 1.
+        stat_key = target_feature.GetFID() + 1

```

Comment:
That's a great point, but I wonder if we need to worry about generalizability here? The formats for these particular vectors are not user-defined; they're constrained by the model itself. The target vector is a shapefile, because it is generated with the ESRI Shapefile driver. In this version, that's via `CreateCopy` (line 557); on main, it's via `Create` (line 564). Meanwhile, the "source" vector is actually the reprojection, which is a Geopackage generated by invoking `pygeoprocessing.reproject_vector` with the GPKG driver (see the `execute` method, lines 413 and 462). So, as long as we don't change those implementation details, we should be OK (and if we do change them, the regression tests should let us know if there's a problem).

What do you think? Happy to go back to adding the fields manually if you still think that's a better approach here.